### ISSUE TITLE
Shipping Labels: Required name field in origin address is empty but still validates until app is relaunched after login

### ISSUE THREAD

**Issue by rachelmcr** [woocommerce/woocommerce-ios#4480]
**Describe the bug**

After logging in to the app, if the app hasn't been relaunched and the user creates a shipping label:

1. The "Name" field in the origin address is not auto-filled.
2. The origin address is still validated even though the Name field is required.
3. The only way to know something is wrong is by tapping on the origin address to see the "Name missing" error there.
4. You can get all the way to the "Carrier and Rates" section before running into a problem (no rates are loaded), and it isn't clear what caused it unless you know to check the origin address.

Error from the API:
```
{
  "error" : "wcc_server_error_response",
  "message" : "Error: The WooCommerce Shipping & Tax server returned: Bad Request child \"origin\" fails because [child \"name\" fails because [\"name\" is not allowed to be empty]] ( 400 )"
}
```

**To Reproduce**

1. Make sure WooCommerce Shipping & Tax is set up on your store, with packages added in the settings.
2. Log in to the app.
3. Open an order detail that is eligible for the shipping label.
4. Tap the button "Create Shipping Label".
5. Tap "Continue" on the Ship From field and notice it is validated.
6. Continue until you get to the Shipping Carrier and Rates field, and notice you get an error there.
7. Go back and tap on the "Ship From" field and notice the "Name missing" error. Enter a name and notice you can now view and select a carrier and rates.

Note that if you force close and relaunch the app at any point, when you create a shipping label the name field is now filled in.

**Screenshots**

Creating a shipping label (no name) | Validating the origin address | Origin address validated | Name missing
-|-|-|-
![Simulator Screen Shot - iPhone 11 Pro - 2021-06-25 at 13 27 06](https://user-images.githubusercontent.com/8658164/123425692-5dd7f700-d5ba-11eb-8187-16b9389284f6.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-06-25 at 13 27 11](https://user-images.githubusercontent.com/8658164/123425696-5f092400-d5ba-11eb-89be-c3bc0453782d.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-06-25 at 13 27 15](https://user-images.githubusercontent.com/8658164/123425701-603a5100-d5ba-11eb-93b5-a00564ca8645.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-06-25 at 13 27 18](https://user-images.githubusercontent.com/8658164/123425710-62041480-d5ba-11eb-949c-da376be1fb66.png)

**Comment by rachelmcr** [comment]
I did a little investigation and made a couple relevant discoveries:

* This issue is actually related to creating a shipping label just after logging in to the app, not for specific user types. (I updated the issue title and description to reflect that.)
* The [remote endpoint ignores the name](https://github.com/Automattic/woocommerce-services/blob/324fcbaac44ae13074eed77bb1a5930d7def5891/classes/class-wc-rest-connect-address-normalization-controller.php#L21) when it validates the address, so we'll have to handle this locally.

This could use a couple fixes:

1. Don't allow the user to continue if the name field is empty. We could build this into `ShippingLabelAction.validateAddress` or just check it within `ShippingLabelFormViewModel.validateAddress`.
2. Make sure the name field is filled out when possible, even if the app hasn't been relaunched. I'm guessing from a quick look that this issue can happen because there aren't any stored account settings when we call `ShippingLabelFormViewModel.getStoredAccountSettings()` the first time after login.
